### ISSUE TITLE
(old) Compiler doesn't pinpoint dialogue and line of incorrectly terminated character constant

### ISSUE THREAD

**Issue by fernewelten** [adventuregamestudio/ags#1326]
**Describe the bug**
In a new game, install a dialogue with one option and the following script:
```
// Dialog script file
@S  // Dialog startup entry point
return
@1
    int j = '';
return
```
Build EXE (F7) and note that in the Output window, the error is given for file `__DialogScripts.asc`, Line 5. It seems that the compiler can't find the __specific__ dialogue that the error is in. Double-clicking on the line in the Output window won't open the specific dialogue.

That's not a problem that the compiler has in general for dialogues. For instance, change the above dialogue so that it says, `int j = Holzschuh;`.
Build EXE (F7) and note that now, the error is correctly given for `Dialog 0`, line 5. Double-clicking on the line in the Output window will open the dialogue and put the cursor on the offending line. So now, the compiler can pinpoint the __specific__ dialogue and the __specific__ line within the dialogue.

**Expected behavior**
The message in the Output window should be accompanied with the specific dialogue in column `File` and the specific line in column `Line`. Doubleclicking on the line should open the dialogue and put the cursor onto the offending line.

**AGS Version**
Tested in 3.5.1 and the current commit in 4.0, with the __old__ compiler.

**To Reproduce**
See above.

**Desktop (please complete the following information):**
Windows 10

**Comment by ivan-mogilko** [comment]
From what I have gathered,

1. The script location is set into the variable called `ccCurScriptName` when NEW_SCRIPT_TOKEN_PREFIX is met (__NEWSCRIPTSTART_).

2. Compilation has two passes: tokenize and writing bytecode.
https://github.com/adventuregamestudio/ags/blob/release-3.5.1/Compiler/script/cs_parser.cpp#L3402

The first error is thrown by `cc_tokenize`, while the second is thrown by later stage.

Apparently, `cc_tokenize` recognizes NEW_SCRIPT_TOKEN_PREFIX, but does not assign new `ccCurScriptName`. This means that any error occuring in `cc_tokenize` will be reported with the `ccCurScriptName` set before a call to tokenize.

**Comment by rofl0r** [comment]
the question is, is this really worth fixing for "old" engine - the editor user at least gets a hint that this is due to something wrong in script.

**Comment by ivan-mogilko** [comment]
> the question is, is this really worth fixing for "old" engine - the editor user at least gets a hint that this is due to something wrong in script.

Dialogs are compiled as a single big script. Some games may have hungreds of these. Finding a missing quote or similar mistake within these may take lots of time.

**Comment by ivan-mogilko** [comment]
I think this should fix the issue: fe3756c5eb35703274aa1b023defc801e29cf558

**Comment by ivan-mogilko** [comment]
Fixed in [3.5.1 patch 1](https://github.com/adventuregamestudio/ags/releases/tag/v.3.5.1.8).
